### PR TITLE
[Ready for review] Added CDN Profiles to fluent

### PR DESCRIPTION
**Pull request by hovsepm**

### CODE REVIEW

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+/**
+ * An immutable client-side representation of an Azure CDN endpoint.
+ */
+@Fluent
+public interface CdnEndpoint extends
+        ExternalChildResource<CdnEndpoint, CdnProfile>,
+        Wrapper<EndpointInner> {
+
+    /**
+     * Get the originHostHeader value.
```
Comment (martinsawicki) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java]: We use the 3rd person declarative mood for member descriptions, so "Gets" not "Get", "Specifies" not "Specify", etc. 
pls scrub the Javadocs to get them consistent with that. thx

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+     *
+     * @return the originHostHeader value
```
Comment (martinsawicki) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java]: this is not a user-facing statement, as it is meaningless. i know this is the wording in the auto-gen'd one. We need to make it sound user-facing. i.e "origin host header" not "originHostHeader" value. pls scrub the javadocs with this in mind.

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+     * Get the isHttpAllowed value.
+     *
+     * @return the isHttpAllowed value
+     */
+    boolean isHttpAllowed();
+
+    /**
+     * Get the isHttpsAllowed value.
+     *
+     * @return the isHttpsAllowed value
```
Comment (martinsawicki) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java]: for the Boolean, one descriptive way would be to say "true is HTTP is allowed, else false"

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java (hunk 1 above)
Comment (martinsawicki) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java]: We use the 3rd person declarative mood for member descriptions, so "Gets" not "Get", "Specifies" not "Specify", etc. 
pls scrub the Javadocs to get them consistent with that. thx

Btw - on parameters properties returning a value, we don't need the description,  just the @return, so:
@return origin host header

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+                 * Specifies the Origin of the CDN Endpoint.
+                 *
+                 * @param originHostName origin hostname.
+                 * @return the next stage of the definition
+                 */
+                DefinitionStages.WithPremiumAttach<ParentT> withPremiumOrigin(String originHostName);
+            }
+        }
+
+        /** The final stage of the CDN profile Standard Akamai or Standard Verizone endpoint definition.
```
Comment (martinsawicki) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java]: Verizon misspelled

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java [hunk 5]
```diff
@@ -0,0 +1,445 @@
...
+            WithStandardAttach<ParentT> withContentTypeToCompress(String contentTypeToCompress);
+            WithStandardAttach<ParentT> withCompressionEnabled(boolean compressionEnabled);
+            WithStandardAttach<ParentT> withQueryStringCachingBehavior(QueryStringCachingBehavior cachingBehavior);
+            WithStandardAttach<ParentT> withGeoFilters(List<GeoFilter> geoFilters);
+            WithStandardAttach<ParentT> withGeoFilter(String relativePath, GeoFilterActions action, CountryISOCode countryCode);
+            WithStandardAttach<ParentT> withGeoFilter(String relativePath, GeoFilterActions action, List<CountryISOCode> countryCodes);
+            WithStandardAttach<ParentT> withCustomDomain(String hostName);
+        }
+
+        /** The final stage of the CDN profile Premium Verizone endpoint definition.
```
Comment (martinsawicki) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java]: pls correct all occurrences of the misspelled "Verizone"

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java [hunk 6]
```diff
@@ -0,0 +1,445 @@
...
+                 * Specifies the Origin of the CDN Endpoint.
+                 *
+                 * @param originName name of the Origin.
+                 * @param originHostName origin hostname.
+                 * @return the next stage of the definition
+                 */
+                UpdateDefinitionStages.WithStandardAttach<ParentT> withOrigin(String originName, String originHostName);
+
+                /**
+                 * Specifies the Origin of the CDN Endpoint.
```
Comment (martinsawicki) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java]: no need to capitalize "origin"

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnProfiles.java [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    void endpointStart(String resourceGroupName, String profileName, String endpointName);
+
+    /**
+     * Stops an existing running CDN endpoint.
+     *
+     * @param resourceGroupName Name of the Resource group within the Azure subscription.
+     * @param profileName Name of the CDN profile which is unique within the resource group.
+     * @param endpointName Name of the endpoint under the profile which is unique globally.
+     */
+    void endpointStop(String resourceGroupName, String profileName, String endpointName);
```
Comment (martinsawicki) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnProfiles.java]: stopEndpoint(...)

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnProfiles.java [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+    /**
+     * Forcibly purges CDN endpoint content.
+     *
+     * @param resourceGroupName Name of the Resource group within the Azure subscription.
+     * @param profileName Name of the CDN profile which is unique within the resource group.
+     * @param endpointName Name of the endpoint under the profile which is unique globally.
+     * @param contentPaths The path to the content to be purged. Can describe a file path or a wild card directory.
+     */
+    void endpointPurgeContent(String resourceGroupName, String profileName, String endpointName, List<String> contentPaths);
```
Comment (martinsawicki) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnProfiles.java]: --> purgeEndpointContent(...)

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/ProfileResourceState.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
+/**
```
Comment (martinsawicki) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/ProfileResourceState.java]: in the future, pls do not include auto-gen'd code with hand-written code in the same PR. Makes full code reviews pretty much impossible :-)

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java (hunk 1 above)
Comment (hovsepm) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java]: Fixed everywhere in CDN public interfaces

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java (hunk 2 above)
Comment (hovsepm) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java]: fixed everywhere.

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java (hunk 3 above)
Comment (hovsepm) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java]: Fixed.

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java (hunk 1 above)
Comment (hovsepm) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java]: Fixed

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java [hunk 10]
```diff
@@ -0,0 +1,445 @@
...
+
+            /**
+             * The stage of the CDN profile endpoint definition allowing to specify the Origin
+             * for CDN Profile with Premium Verizon sku.
+             *
+             * @param <ParentT> the return type of {@link AttachablePremium#attach()}
+             */
+            interface PremiumEndpoint<ParentT> {
+                /**
+                 * Specifies the Origin of the CDN Endpoint.
```
Comment (hovsepm) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java]: Fixed

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java (hunk 4 above)
Comment (hovsepm) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java]: Fixed

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java (hunk 5 above)
Comment (hovsepm) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java]: Fixed

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java (hunk 6 above)
Comment (hovsepm) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java]: Fixed

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnProfiles.java [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+    PagedList<Operation> listOperations();
+
+    /**
+     * Starts an existing stopped CDN endpoint.
+     *
+     * @param resourceGroupName Name of the Resource group within the Azure subscription.
+     * @param profileName Name of the CDN profile which is unique within the resource group.
+     * @param endpointName Name of the endpoint under the profile which is unique globally.
+     */
+    void endpointStart(String resourceGroupName, String profileName, String endpointName);
```
Comment (hovsepm) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnProfiles.java]: Fixed

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnProfiles.java (hunk 7 above)
Comment (hovsepm) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnProfiles.java]: Fixed

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnProfiles.java (hunk 8 above)
Comment (hovsepm) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnProfiles.java]: Fixed

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnProfiles.java [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+
+    /**
+     * Forcibly pre-loads CDN endpoint content. Available for Verizon Profiles.
+     *
+     * @param resourceGroupName Name of the Resource group within the Azure subscription.
+     * @param profileName Name of the CDN profile which is unique within the resource group.
+     * @param endpointName Name of the endpoint under the profile which is unique globally.
+     * @param contentPaths The path to the content to be loaded. Should describe a file path.
+     */
+    void endpointLoadContent(String resourceGroupName, String profileName, String endpointName, List<String> contentPaths);
```
Comment (hovsepm) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnProfiles.java]: Fixed

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/ProfileResourceState.java (hunk 9 above)
Comment (hovsepm) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/ProfileResourceState.java]: ok

**Comment by hovsepm** [comment]
Working on rebase to split generated and handcrafted files in two separate commits that will make code review easier...

**Pull request closed without merging**